pilot/pkg/model/push_context.go

```diff
@@ -481,36 +495,53 @@ func (ps *PushContext) GetAllSidecarScopes() map[string][]*SidecarScope {
 }
 
 // DestinationRule returns a destination rule for a service name in a given domain.
-func (ps *PushContext) DestinationRule(proxy *Proxy, hostname Hostname) *Config {
+func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *Config {
 	// If proxy has a sidecar scope that is user supplied, then get the destination rules from the sidecar scope
```

Comment:
To double check: the legacy case ( no Sidecar ) will hit this method and fall trough , while 'sidecar=true' will hit the first if, and return the per-proxy pre-computed DR ?

Reply:
May be worth adding this to comments ( and if false - adding the opposite :-)